crates/fsrt/src/test.rs

```diff
@@ -0,0 +1,224 @@
+use crate::{forge_project::ForgeProjectTrait, scan_directory, Args};
+use clap::Parser;
+use forge_analyzer::{definitions::PackageData, reporter::Vulnerability};
+use forge_loader::manifest::{ForgeManifest, FunctionMod};
+use std::fmt;
+use std::{
+    collections::{HashMap, HashSet},
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+use swc_core::common::sync::Lrc;
+use swc_core::{
+    common::{FileName, SourceFile, SourceMap},
+    ecma::visit::FoldWith,
+};
+use tracing::debug;
+
+#[derive(Clone)]
+pub(crate) struct MockForgeProject<'a> {
+    pub files_name_to_source: HashMap<PathBuf, Arc<SourceFile>>,
+    pub test_manifest: ForgeManifest<'a>,
+    pub cm: Lrc<SourceMap>,
+}
+
+impl fmt::Debug for MockForgeProject<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Mock Forge Project {:?}", self.files_name_to_source)
+    }
+}
+
+#[allow(dead_code)]
+impl MockForgeProject<'_> {
+    pub fn files_from_string(string: String) -> Self {
+        let forge_manifest = ForgeManifest::create_manifest_with_func_mod(FunctionMod {
+            key: "main",
+            handler: "index.run",
+            providers: None,
+        });
+
+        let mut mock_forge_project = MockForgeProject {
+            files_name_to_source: HashMap::new(),
+            test_manifest: forge_manifest,
+            cm: Arc::default(),
+        };
+
+        let different_files = string
+            .split("//")
+            .map(|f| f.replace("//", "").trim().to_string())
+            .filter(|file| !file.is_empty());
+        for file in different_files {
+            let (file_name, file_source) = file.split_once('\n').unwrap();
+            mock_forge_project.add_file(file_name.trim(), file_source.into());
+        }
+
+        mock_forge_project
+    }
+
+    fn add_file(&mut self, p: impl Into<PathBuf>, source: String) {
+        let file_name = p.into();
+        let source_file = self
+            .cm
+            .new_source_file(FileName::Real(file_name.clone()), source);
+
+        self.files_name_to_source.insert(file_name, source_file);
+    }
+}
+
+impl<'a> ForgeProjectTrait<'a> for MockForgeProject<'a> {
+    fn load_file(&self, p: impl AsRef<Path>, _: Arc<SourceMap>) -> Arc<SourceFile> {
+        self.files_name_to_source.get(p.as_ref()).unwrap().clone()
+    }
+
+    fn get_paths(&self) -> HashSet<PathBuf> {
+        self.files_name_to_source
+            .keys()
+            .map(|file| file.into())
+            .collect::<HashSet<_>>()
+    }
+
+    #[allow(dead_code)]
+    fn get_secret_packages(&self) -> Vec<PackageData> {
+        vec![]
+    }
+
+    fn get_manifest(&self) -> ForgeManifest<'_> {
+        self.test_manifest.clone()
+    }
+
+    fn with_files_and_sourceroot<
```

Comment:
Is there a reason why this doesn't use the default implementation?

Reply:
nope. I removed it